### PR TITLE
2_trees_mining

### PR DESCRIPTION
**Pull request by Oris96**

### CODE REVIEW

File: Source/TryToSurvive/Models/Trees/ChoppableTree.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		SetNumberOfLogs();
+	}
+}
+
+void AChoppableTree::SetNumberOfLogs()
+{
+	AmountOfLogs = FMath::RandRange(4, 6);
+}
+
+void AChoppableTree::ReceiveDamage(float damage)
```
Comment (MrSbkz) [inline Source/TryToSurvive/Models/Trees/ChoppableTree.cpp:37]: Please, move this logic to the AActorWithHealthBase class

File: Source/TryToSurvive/Models/Trees/LogActor.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include "LogActor.h"
+
+ALogActor::ALogActor()
+{
+	CapsuleComponent = CreateDefaultSubobject<UCapsuleComponent>("CollisionComponent");
+	CapsuleComponent->SetupAttachment(MeshComponent);
+
+	CapsuleComponent->SetCapsuleSize(12.0f, 115.0f);
```
Comment (MrSbkz) [inline Source/TryToSurvive/Models/Trees/LogActor.cpp:8]: Move collision setup into BeginPlay and set it by mesh extend

File: Source/TryToSurvive/Models/Trees/StumpActor.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#include "StumpActor.h"
+
+AStumpActor::AStumpActor()
+{
+	SetRootComponent(MeshComponent);
+
+	BoxComponent = CreateDefaultSubobject<UBoxComponent>("CollisionComponent");
+	BoxComponent->SetupAttachment(MeshComponent);
+
+	BoxComponent->SetBoxExtent(FVector(22.0f, 22.0f, 22.0f));
```
Comment (MrSbkz) [inline Source/TryToSurvive/Models/Trees/StumpActor.cpp:10]: Move collision setup into BeginPlay and set it by mesh extend

File: Source/TryToSurvive/TryToSurviveCharacter.h [hunk 4]
```diff
@@ -18,18 +23,21 @@ class ATryToSurviveCharacter : public ACharacter
...
-	USkeletalMeshComponent* Mesh1P;
-
 	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Category=Components)
 	UBuildingComponent* BuildingComponent;
 
+	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = Camera)
+	USpringArmComponent* SpringArmComponent;
+
+	UPROPERTY(EditAnywhere, BlueprintReadOnly, Category = Animation)
+	UAnimMontage* ChopAnimation;
```
Comment (MrSbkz) [inline Source/TryToSurvive/TryToSurviveCharacter.h:39]: Maybe it makes sense to add generic animation(CurrentAnimation)

File: Source/TryToSurvive/TryToSurviveCharacter.h [hunk 5]
```diff
@@ -41,18 +49,35 @@ class ATryToSurviveCharacter : public ACharacter
...
+	bool isEnabledAttack = true;
+
+	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = Camera)
+	UCameraComponent* ThirdPersonCameraComponent;
+
+	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = Camera, meta = (AllowPrivateAccess = "true"))
+	UCameraComponent* FirstPersonCameraComponent;
+
 private:
+	void Chop();
```
Comment (MrSbkz) [inline Source/TryToSurvive/TryToSurviveCharacter.h:69]: Create a separate attack component

Comment (Oris96) [reply Source/TryToSurvive/TryToSurviveCharacter.h:69]: This is personal action not weapon

**Review by MrSbkz — APPROVED** [review]

**Pull request merged**